### ISSUE TITLE
Dependencies not recorded in pom.xml

### ISSUE THREAD

**Issue by sebastic** [simonpoole/OpeningHoursParser#67]
Per the discussion in [JOSM Trac #21140](https://josm.openstreetmap.de/ticket/21140), the org.jetbrains.annotations dependency is not recorded in [pom.xml](https://repo1.maven.org/maven2/ch/poole/OpeningHoursParser/0.23.4/OpeningHoursParser-0.23.4.pom).

It seems that the dependencies need to be explicitly configured in the `pom{}` section in `build.gradle` which it doesn't have currently:
```
            pom {
                name = artifactId
                description = 'This is a simple parser for OSM "opening_hours" tags'    
                url = 'https://github.com/simonpoole/OpeningHoursParser'
                scm {
                    url = repoUrl
                }
                licenses {
                    license {
                        name = spdxId
                        url = 'https://raw.githubusercontent.com/simonpoole/OpeningHoursParser/master/LICENCE.txt'
                    }
                }
                developers {
                    developer {
                        name = 'Simon Poole'
                    }
                }
            }
```
According to the [documentation](https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:resolved_dependencies) `versionMapping` can be used to use the resolved version as an alternative to declared versions.

**Comment by simonpoole** [comment]
Good morning Bas

This is somewhat an unexpected expected issue, because the dependency is compile time only the publishing plugin doesn't add it at all (because it isn't a runtime dependency). 

Looking in to how to fudge this right now, but unluckily pom scopes don't actually have an 1-1 equivalent for 'compile", so likely I'll change it back to a regular dependency.

**Comment by simonpoole** [comment]
@sebastic would this work for you?

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <!-- This module was also published with a richer model, Gradle metadata,  -->
  <!-- which should be used instead. Do not delete the following line which  -->
  <!-- is to indicate to Gradle or any Gradle module metadata file consumer  -->
  <!-- that they should prefer consuming it instead. -->
  <!-- do_not_remove: published-with-gradle-metadata -->
  <modelVersion>4.0.0</modelVersion>
  <groupId>ch.poole</groupId>
  <artifactId>OpeningHoursParser</artifactId>
  <version>0.23.4</version>
  <packaging>pom</packaging>
  <name>OpeningHoursParser</name>
  <description>This is a simple parser for OSM "opening_hours" tags</description>
  <url>https://github.com/simonpoole/OpeningHoursParser</url>
  <licenses>
    <license>
      <name>MIT</name>
      <url>https://raw.githubusercontent.com/simonpoole/OpeningHoursParser/master/LICENCE.txt</url>
    </license>
  </licenses>
  <developers>
    <developer>
      <name>Simon Poole</name>
    </developer>
  </developers>
  <scm>
    <url>https://github.com/simonpoole/OpeningHoursParser.git</url>
  </scm>
  <dependencies>
    <dependency>
      <groupId>org.jetbrains</groupId>
      <artifactId>annotations</artifactId>
      <version>19.0.0</version>
      <optional>true</optional>
    </dependency>
  </dependencies>
</project>
```

This requires directly modifying the xml in the gradle script and is ugly^2 but probably is a half workable solution.

**Comment by sebastic** [comment]
That should work, but I don't understand why gradle can't generate that. Does using declared versions not work either? That may be preferable over modifying the XML.

**Comment by simonpoole** [comment]
> 
> 
> That should work, but I don't understand why gradle can't generate that. Does using declared versions not work either? That may be preferable over modifying the XML.

There are a number of issues, but essentially I wanted to avoid requiring users of the binary artifact to have to include a totally unnecessary dependency, so I declared the annotations as 'compileOnly' aka only used to compile the code, but no runtime dependency.

The gradle maven-publish plugin doesn't include such dependencies in the pom file and even it did, there is no equivalent scope for "compileOnly", Adding the 'optional' element makes both gradle and maven ignore the dependency so should have the same effect in this case, but it is rather hackish.

PS: I couldn't get versionMapping to work in any reasonable way, I had had some hope that I could use it to get around the plugin ignoring compileOnly dependency.

**Comment by simonpoole** [comment]
Fixed in 0.24.0 https://github.com/simonpoole/OpeningHoursParser/releases/tag/0.24.0
